src/main/java/org/folio/circulation/domain/LoanRepository.java

```diff
@@ -335,4 +335,24 @@ private Request matchLoansToRequest(
       .withLoan(loanMap.getOrDefault(request.getItemId(), null));
   }
 
+  public CompletableFuture<Result<MultipleRecords<Loan>>> findOpenLoansByUserIdAndLoanPolicyId(
+    LoanAndRelatedRecords loanAndRelatedRecords) {
+    String userId = loanAndRelatedRecords.getLoan().getUser().getId();
+    String loanPolicyId = loanAndRelatedRecords.getLoan().getLoanPolicy().getId();
+    final Result<CqlQuery> statusQuery = getStatusCQLQuery("Open");
+    final Result<CqlQuery> userIdQuery = exactMatch("userId", userId);
+    final Result<CqlQuery> loanPolicyIdQuery = exactMatch("loanPolicyId", loanPolicyId);
+    Result<CqlQuery> cqlQueryResult = statusQuery
+      .combine(userIdQuery, CqlQuery::and)
+      .combine(loanPolicyIdQuery, CqlQuery::and);
+
+    return queryLoanStorage(Integer.MAX_VALUE, cqlQueryResult);
+  }
+
+  public CompletableFuture<Result<MultipleRecords<Loan>>> findOpenLoansByUserIdAndLoanPolicyIdWithItem(
```

Comment:
What are the circumstances where the item wouldn't be wanted?

Reply:
Item wouldn't be wanted when material type and loan type are not present in circulation rule. In that case it doesn't make sense to fetch items because we will not use them for filtering.